site/docs/integrations/next.md

```diff
@@ -66,3 +66,13 @@ const withVanillaExtract = createVanillaExtractPlugin({
 ```
 
 Each integration will set a default value based on the configuration options passed to the bundler.
+
+## CSS load order issues
+
+There are some known issues with Next.js css load order and/or duplicate css.
+
+- When client-side navigating, new css is injected into the head, but unused css isn't removed. This means if route B has a component that applies some global css to a component also in route A, upon returning to route A, those styles from route B will still be applied.
+- When using App router, css can be duplicated which could cause unexpected overrides.
+
+[next.js #51030](https://github.com/vercel/next.js/issues/51030)
+[next.js #40080](https://github.com/vercel/next.js/issues/40080)
```

Comment:
@askoufis this is how I see it from my testing, but maybe I'm missing something.
There are quite a few issues in the next.js repo regarding css loading order etc., although it doesn't seem as bad with vanilla extract as it is with css modules for example. 

Here's a repro demonstrating the lingering css overrides. This seems to happen in every mode/router aside from pages router in production.
https://codesandbox.io/p/sandbox/next-51030-forked-phqdsr

To some degree it's kind of expected behavior? I guess we need to keep tabs on any future improvements on the next side.